Framework/PythonInterface/plugins/algorithms/IntegratePeaksShoeboxTOF.py

```diff
@@ -0,0 +1,687 @@
+        get_nbins_from_b2bexp_params = self.getProperty("GetNBinsFromBackToBackParams").value
+        nfwhm = self.getProperty("NFWHM").value
+        nshoebox = self.getProperty("NShoeboxInWindow").value
+        # shoebox optimisation
+        do_optimise_shoebox = self.getProperty("OptimiseShoebox").value
+        weak_peak_strategy = self.getProperty("WeakPeakStrategy").value
+        weak_peak_threshold = self.getProperty("WeakPeakThreshold").value
+        # validation
+        integrate_on_edge = self.getProperty("IntegrateIfOnEdge").value
+        nrows_edge = self.getProperty("NRowsEdge").value
+        ncols_edge = self.getProperty("NColsEdge").value
+        # corrections
+        do_lorz_cor = self.getProperty("LorentzCorrection").value
+        # saving file
+        output_file = self.getProperty("OutputFile").value
+
+        # create output table workspace
+        peaks = self.exec_child_alg("CloneWorkspace", InputWorkspace=peaks, OutputWorkspace="out_peaks")
+
+        array_converter = InstrumentArrayConverter(ws)
+        weak_peaks_list = []
+        ipks_strong = []
+        results = np.full(peaks.getNumberPeaks(), None)
+        prog_reporter = Progress(self, start=0.0, end=1.0, nreports=peaks.getNumberPeaks())
+        for ipk, peak in enumerate(peaks):
+            prog_reporter.report("Integrating")
+            status = PEAK_STATUS.NO_PEAK
+            intens, sigma = 0.0, 0.0
+
+            detid = peak.getDetectorID()
+            bank_name = peaks.column("BankName")[ipk]
+            pk_tof = peak.getTOF()
+
+            # check TOF is in limits of x-axis
+            xdim = ws.getXDimension()
+            if xdim.getMinimum() < pk_tof < xdim.getMaximum():
+                # get shoebox kernel for initial integration
+                nrows = self.getProperty("NRows").value  # get these inside loop as overwritten if shoebox optimised
+                ncols = self.getProperty("NCols").value
+                ispec = ws.getIndicesFromDetectorIDs([detid])[0]
+                itof = ws.yIndexOfX(pk_tof, ispec)
+                bin_width = np.diff(ws.readX(ispec)[itof : itof + 2])[0]  # used later to scale intensity
+                if get_nbins_from_b2bexp_params:
+                    fwhm = get_fwhm_from_back_to_back_params(peak, ws, detid)
+                    nbins = max(3, int(nfwhm * fwhm / bin_width)) if fwhm is not None else self.getProperty("NBins").value
+                    nbins = round_up_to_odd_number(nbins)
+                else:
+                    nbins = self.getProperty("NBins").value
+                kernel = make_kernel(nrows, ncols, nbins)
+
+                # get data array and crop
+                peak_data = array_converter.get_peak_data(
+                    peak, detid, bank_name, nshoebox * kernel.shape[0], nshoebox * kernel.shape[1], nrows_edge, ncols_edge
+                )
+                x, y, esq, ispecs = get_and_clip_data_arrays(ws, peak_data, pk_tof, ispec, kernel, nshoebox)
```

Comment:
When reviewing the function, I got a bit confused by the meaning of x and y. Since they don't have the same shape and are not one-dimensional arrays. Perhaps is better to rename x to x_tof? and esq to and y to other more descriptive names? What do you think?

Reply:
Yep, I agree it could be confusing. The rectangular detector class uses X and Y for the two coordinates (row/column), but here i also use x to denote the x-axis of the workspace, in this case TOF (I use `x` instead of `tof` did this because at some point it could be generalised to work with d-spacing). I'll have a go at clearing up some of the confusion!